File: Day1/Day1.py
def problem1(file: str) -> str:
    """
    Solution to problem1
    """
    lines = []
    with open(file, "r") as f:
        for line in f:
            lines.append(line.strip())

    # Create two variables, one to hold the highest amt of calories, the other to temporarily hold the amount of calories the current elf is holding
    max = 0
    temp = 0

    # Loop through all items
    for item in lines:
        # If the item is not blank, we can keep adding it to temp
        if item.strip() != "":
            temp += int(item)
        # Otherwise we see if it's bigger than the max, and set temp to 0
        else:
            if temp > max:
                max = temp
            temp = 0

    return max


def problem2(file: str) -> str:
    """
    Solution to problem2
    """
    lines = []
    with open(file, "r") as f:
        for line in f:
            lines.append(line.strip())

    # First we create a list to hold the top three elves calories
    top = []
    # We create another temp var
    temp = 0

    # Then we use the same logic, with a bit of tweaking
    for line in lines:
        if line.strip() != "":
            temp += int(line)
        else:
            # If top has less than 3 calories
            if len(top) < 3:
                top.append(temp)
            else:
                # Keep a "leaderboard" of calories
                if temp > top[2]:
                    top.append(temp)
                    temp = top.pop(2)
                if temp > top[1]:
                    top.append(temp)
                    temp = top.pop(1)
                if temp > top[0]:
                    top.pop(0)
                    top.append(temp)
            temp = 0

    return top[0] + top[1] + top[2]


print(problem1("data.txt"))
print(problem2("data.txt"))

File: Day1/Day1.java
import java.io.File;
import java.io.FileNotFoundException;
import java.util.ArrayList;
import java.util.Scanner;

class Day1 {

    public static int problem1(String filename) throws FileNotFoundException
    {
        File f = new File(filename);
        Scanner s = new Scanner(f);
        try
        {
            ArrayList<String> lines = new ArrayList<String>();
            while (s.hasNextLine())
            {
                lines.add(s.nextLine().strip());
            }

            // Create 2 vars, one to hold biggest amt, the other to temporarily hold top values
            int biggest = 0;
            int temp = 0;

            // Loop through all strings
            for (String line: lines) {
                // If we don't meet a blank, continue adding to temp
                if (line.strip() != "") {
                    temp += Integer.valueOf(line);
                } else {
                    // Check if we need to change biggest to temp
                    if (temp > biggest) {
                        biggest = temp;
                    }
                    temp = 0;
                }
            }

            return biggest;
        }
        finally
        {
            s.close();
        }
    }

    public static int problem2(String filename) throws FileNotFoundException
    {
        File f = new File(filename);
        Scanner s = new Scanner(f);
        try
        {
            ArrayList<String> lines = new ArrayList<String>();
            while (s.hasNextLine())
            {
                lines.add(s.nextLine().strip());
            }

            // Create 2 vars, one array of size 3 to hold top 3 values, other to temp hold values
            int[] top = {0, 0, 0};
            int temp = 0;

            // Loop through all values
            for (String line: lines) {
                if (line.strip() != "") {
                    // Same as before, keep adding to temp until we find a blank
                    temp += Integer.valueOf(line);
                } else {
                    // Loop through all values of top, if we find a value that is higher, then set it. continue to keep track of all variables
                    int secondTemp;
                    if (temp > top[2]) {
                        secondTemp = top[2];
                        top[2] = temp;
                        temp = secondTemp;
                    }
                    if (temp > top[1]) {
                        secondTemp = top[1];
                        top[1] = temp;
                        temp = secondTemp;
                    }
                    if (temp > top[0]) {
                        secondTemp = top[0];
                        top[0] = temp;
                        temp = secondTemp;
                    }
                    temp = 0;
                }
            }
            
            return top[0] + top[1] + top[2];
        }
        finally
        {
            s.close();
        }
    }
    
    public static void main(String[] args) {
        try {
            System.out.println(problem1("data.txt"));
            System.out.println(problem2("data.txt"));
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}